timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/cron/BulkPersistentFetchJob.java

```diff
@@ -1,58 +1,58 @@
+
 import static com.datorama.oss.timbermill.common.ElasticsearchUtil.CLIENT;
-import static com.datorama.oss.timbermill.common.ElasticsearchUtil.DISK_HANDLER;
+import static com.datorama.oss.timbermill.common.ElasticsearchUtil.PERSISTENCE_HANDLER;
 
 @DisallowConcurrentExecution
 public class BulkPersistentFetchJob implements Job {
 
-	private static final Logger LOG = LoggerFactory.getLogger(BulkPersistentFetchJob.class);
-
-	@Override public void execute(JobExecutionContext context) {
-		DiskHandler diskHandler = (DiskHandler) context.getJobDetail().getJobDataMap().get(DISK_HANDLER);
-		if (diskHandler != null) {
-			Timer.Started start = KamonConstants.BULK_FETCH_JOB_LATENCY.withoutTags().start();
-			String flowId = "Failed Bulk Persistent Fetch Job - " + UUID.randomUUID().toString();
-			MDC.put("id", flowId);
-			LOG.info("Failed Bulks Persistent Fetch Job started.");
-			ElasticsearchClient es = (ElasticsearchClient) context.getJobDetail().getJobDataMap().get(CLIENT);
-			retryFailedRequestsFromDisk(es, diskHandler);
-			LOG.info("Failed Bulks Persistent Fetch Job ended.");
-			start.stop();
-		}
-	}
-
-	private static void retryFailedRequestsFromDisk(ElasticsearchClient es, DiskHandler diskHandler) {
-		String flowId = MDC.get("id");
-		while(diskHandler.hasFailedBulks()){
-			LOG.info("#### Retry Failed-Requests From Disk Start ####");
-			List<DbBulkRequest> failedRequestsFromDisk = diskHandler.fetchAndDeleteFailedBulks();
-			int failedRequests = failedRequestsFromDisk.stream().mapToInt(DbBulkRequest::numOfActions).sum();
-
-			int successBulks = 0;
-			int bulkNum = 0;
-			for (DbBulkRequest failedDbBulkRequest : failedRequestsFromDisk) {
-				successBulks += es.sendDbFailedBulkRequest(failedDbBulkRequest, flowId, bulkNum);
-				bulkNum++;
-			}
-			LOG.info("#### Retry Failed-Requests From Disk End ({}/{} fetched bulks re-processed successfully) ####", successBulks, failedRequests);
-		}
-	}
+    private static final Logger LOG = LoggerFactory.getLogger(BulkPersistentFetchJob.class);
+
+    @Override
+    public void execute(JobExecutionContext context) {
+        PersistenceHandler persistenceHandler = (PersistenceHandler) context.getJobDetail().getJobDataMap().get(PERSISTENCE_HANDLER);
+        if (persistenceHandler != null) {
+            KamonConstants.CURRENT_DATA_IN_DB_GAUGE.withTag("type", "failed_bulks_amount").update(persistenceHandler.failedBulksAmount());
+            Timer.Started start = KamonConstants.BULK_FETCH_JOB_LATENCY.withoutTags().start();
+            String flowId = "Failed Bulk Persistent Fetch Job - " + UUID.randomUUID().toString();
+            MDC.put("id", flowId);
+            LOG.info("Failed Bulks Persistent Fetch Job started.");
+            ElasticsearchClient es = (ElasticsearchClient) context.getJobDetail().getJobDataMap().get(CLIENT);
+            retryFailedRequests(es, persistenceHandler);
+            LOG.info("Failed Bulks Persistent Fetch Job ended.");
+            start.stop();
+        }
```

Comment:
I considered adding a lock here -
there is no point that each pod would fetch the same queue.
Though it's specifically to the Redis implementation so maybe it's not such a good idea